service/gamelift/serializers.go

```diff
@@ -2395,6 +2395,15 @@ func (m *awsAwsjson11_serializeOpDescribeVpcPeeringAuthorizations) HandleSeriali
 	httpBindingEncoder.SetHeader("Content-Type").String("application/x-amz-json-1.1")
 	httpBindingEncoder.SetHeader("X-Amz-Target").String("GameLift.DescribeVpcPeeringAuthorizations")
 
+	jsonEncoder := smithyjson.NewEncoder()
+	if err := awsAwsjson11_serializeOpDocumentDescribeVpcPeeringAuthorizationsInput(input, jsonEncoder.Value); err != nil {
+		return out, metadata, &smithy.SerializationError{Err: err}
+	}
+
+	if request, err = request.SetStream(bytes.NewReader(jsonEncoder.Bytes())); err != nil {
+		return out, metadata, &smithy.SerializationError{Err: err}
+	}
+
```

Comment:
Is this intentional? Asking as on a previous comment, you responded you would be removing these. 

I am fine with having them for now.  Think they can be replaced with 

```
	if request, err = request.SetStream(strings.NewReader(`{}`)); err != nil {
		return out, metadata, &smithy.SerializationError{Err: err}
	}
```

If we make the change, would also want to consider not generating dead code.

Reply:
So I left using the generated serializer function if there was a modeled input. This latest updates the generator to only have the short cut path if there was nothing modeled. This also has consistent behavior across all of the protocols too, (e.g. REST-JSON/REST-XML). The reasoning behind this was more for consistent behavior across the generator. As in, all protocols that have stub synethentic clones (aka unmodeled input/output) the generated code will short cut and not render document serializers.

To not generate document (de)serializers for empty input/output shapes is a bit more complicated as that is spread across both smithy-go and sdk's codegen.